src/btree/bt_handle.c

```diff
@@ -722,23 +721,32 @@ __btree_tree_open_empty(WT_SESSION_IMPL *session, bool creation)
  *     Create an empty leaf page.
  */
 int
-__wt_btree_new_leaf_page(WT_SESSION_IMPL *session, WT_PAGE **pagep)
+__wt_btree_new_leaf_page(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_BTREE *btree;
 
     btree = S2BT(session);
 
     switch (btree->type) {
     case BTREE_COL_FIX:
-        WT_RET(__wt_page_alloc(session, WT_PAGE_COL_FIX, 0, false, pagep));
+        WT_RET(__wt_page_alloc(session, WT_PAGE_COL_FIX, 0, false, &ref->page));
         break;
     case BTREE_COL_VAR:
-        WT_RET(__wt_page_alloc(session, WT_PAGE_COL_VAR, 0, false, pagep));
+        WT_RET(__wt_page_alloc(session, WT_PAGE_COL_VAR, 0, false, &ref->page));
         break;
     case BTREE_ROW:
-        WT_RET(__wt_page_alloc(session, WT_PAGE_ROW_LEAF, 0, false, pagep));
+        WT_RET(__wt_page_alloc(session, WT_PAGE_ROW_LEAF, 0, false, &ref->page));
         break;
     }
+
+    /*
```

Comment:
What was happening before this change? You had a page with key-value pairs at the leaf spot of the tree but its type was internal?

Reply:
A deleted internal node (so, a `WT_REF` with an internal type, where everything had been removed), was being re-created (if stuff is removed, but then the app inserts into that empty name space, we have to re-create it). When we re-create those subtrees, we create a single leaf page rather than building out a subtree, and so we ended up with a `WT_REF` that had a type that didn't match the underlying page.